src/ui/layout/mct-tree.vue

```diff
@@ -10,107 +13,452 @@
         />
     </div>
 
-    <!-- loading -->
-    <div
-        v-if="isLoading"
-        class="c-tree-and-search__loading loading"
-    ></div>
-    <!-- end loading -->
-
     <div
-        v-if="(allTreeItems.length === 0) || (searchValue && filteredTreeItems.length === 0)"
+        v-if="(searchValue && allTreeItems.length === 0 && !isLoading) || (searchValue && searchResultItems.length === 0)"
         class="c-tree-and-search__no-results"
     >
         No results found
     </div>
 
     <!-- main tree -->
     <ul
-        v-if="!isLoading"
-        v-show="!searchValue"
+        ref="mainTree"
         class="c-tree-and-search__tree c-tree"
     >
-        <tree-item
-            v-for="treeItem in allTreeItems"
-            :key="treeItem.id"
-            :node="treeItem"
-        />
+        <!-- ancestors -->
+        <div v-if="!activeSearch">
+            <tree-item
+                v-for="(ancestor, index) in ancestors"
+                :key="ancestor.id"
+                :node="ancestor"
+                :show-up="index < ancestors.length - 1"
+                :show-down="false"
+                :left-offset="index * 10 + 'px'"
+                :emit-height="getChildHeight"
+                @emittedHeight="setChildHeight"
+                @resetTree="handleReset"
+            />
+            <!-- loading -->
+            <li
+                v-if="isLoading"
+                class="c-tree__item c-tree-and-search__loading loading"
+            >
+                <span class="c-tree__item__label">Loading...</span>
+            </li>
+            <!-- end loading -->
+        </div>
+        <!-- currently viewed children -->
+        <transition
```

Comment:
This is cool

Reply:
Yeah I was stoked when I found that. Really came through here.